### PR TITLE
Formula widgets: Optimize and cleanup DB query

### PR DESCRIPTION
**Pull request by Algunenano**
Optimize formula queries.

Since the formula and histogram changes (#797) are going to be kind of large I've decided to break the changes in several PRs (not sure if 2 or 3 (numeric histogram, date histogram, formula)).

### CODE REVIEW

File: lib/cartodb/models/dataview/formula.js [hunk 1]
```diff
@@ -1,34 +1,21 @@
...
-const filterOutSpecialNumericValuesTpl = ctx => `
-    WHERE
-        ${ctx.column} != 'infinity'::float
-    AND
-        ${ctx.column} != '-infinity'::float
-    AND
-        ${ctx.column} != 'NaN'::float
-`;
-
+/* jshint ignore:start */
```
Comment (rafatower) [inline lib/cartodb/models/dataview/formula.js]: This is kind of cheating and should be avoided. I guess jshint was complaining about something but don't know what that was. Cyclomatic complexity by any chance?

File: lib/cartodb/models/dataview/formula.js [hunk 2]
```diff
@@ -1,34 +1,21 @@
...
 const formulaQueryTpl = ctx => `
-    SELECT
-        ${ctx.operation}(${ctx.column}) AS result,
-        (SELECT count(1) FROM (${ctx.query}) _cdb_formula_nulls WHERE ${ctx.column} IS NULL) AS nulls_count
-        ${ctx.isFloatColumn ? `,(${countInfinitiesQueryTpl(ctx)}) AS infinities_count` : ''}
-        ${ctx.isFloatColumn ? `,(${countNansQueryTpl(ctx)}) AS nans_count` : ''}
-    FROM (${ctx.query}) __cdb_formula
-    ${ctx.isFloatColumn && ctx.operation !== 'count' ? `${filterOutSpecialNumericValuesTpl(ctx)}` : ''}
-`;
+${!ctx.isFloatColumn ?
```
Comment (rafatower) [inline lib/cartodb/models/dataview/formula.js]: I'd actually use 2 different templates

File: lib/cartodb/models/dataview/overviews/formula.js [hunk 3]
```diff
@@ -6,50 +6,35 @@ var dot = require('dot');
 dot.templateSettings.strip = false;
 
 var formulaQueryTpls = {
+/* jshint ignore:start */
```
Comment (rafatower) [inline lib/cartodb/models/dataview/overviews/formula.js]: same here

File: lib/cartodb/models/dataview/formula.js (hunk 1 above)
Comment (Algunenano) [reply lib/cartodb/models/dataview/formula.js]: No, long lines and I didn't want to mess up the SQL query structure to make the js linter happy.

File: lib/cartodb/models/dataview/formula.js (hunk 2 above)
Comment (Algunenano) [reply lib/cartodb/models/dataview/formula.js]: I'll see what I can do

File: lib/cartodb/models/dataview/formula.js (hunk 1 above)
Comment (rafatower) [reply lib/cartodb/models/dataview/formula.js]: I totally agree with you, better not to split into several lines for the sake of readability.

You can just add a comment to state that.

**Comment by Algunenano** [comment]
When working in the histograms some stuff has come up that could be put in common, so I'll close this and  open a single PR with several commits.

**Pull request closed without merging**